src/main/java/services/RestoreFactory.java

```diff
@@ -337,22 +329,38 @@ private InputStream getRestoreXmlHelper(String restoreUrl, HqAuth auth) {
         return stream;
     }
 
-    public static String getRestoreUrl(String host, String domain, boolean overwriteCache){
-        String url = host + "/a/" + domain + "/phone/restore/?version=2.0";
-        if (overwriteCache) {
-            url += "&overwrite_cache=true";
+    private String getSyncToken(String username) {
+
+        if (username == null) {
+            return null;
+        }
+        SqliteIndexedStorageUtility<User> storage = getSqlSandbox().getUserStorage();
+        Vector<Integer> users = storage.getIDsForValue(User.META_USERNAME, username);
+        //should be exactly one user
+        if (users.size() != 1) {
+            return null;
         }
-        return url;
-    }
 
-    public String getRestoreUrl(String host, String domain, String username, boolean overwriteCache) {
-        String url = host + "/a/" + domain + "/phone/restore/?as=" + username + "@" +
-                domain + ".commcarehq.org&version=2.0";
+        return storage.getMetaDataFieldForRecord(users.firstElement(), User.META_SYNC_TOKEN);
+    }
 
+    public String getRestoreUrl(boolean overwriteCache) {
+        StringBuilder builder = new StringBuilder();
+        builder.append(host);
+        builder.append("/a/");
+        builder.append(domain);
+        builder.append("/phone/restore/?version=2.0");
         if (overwriteCache) {
-            url += "&overwrite_cache=true";
+            builder.append("&overwrite_cache=true");
```

Comment:
the reason this was an issue before was that we didn't actually use sync log ids. restores would then get cached for an hour if they took too long to load. this caused users to not get back a fresh restore and we were getting many tickets about it from icds since their workflow requires that you hit sync at the end of doing some form entry